packages/react-native/src/lib/environment-state.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-console -- logging required for error logging */
+// shared functions for environment and person state(s)
+import { type TJsEnvironmentState, type TJsEnvironmentSyncParams } from "@formbricks/types/js";
+import { err } from "../../../js-core/src/lib/errors";
+import { Logger } from "../../../js-core/src/lib/logger";
+import { RNConfig } from "./config";
+import { filterSurveys } from "./utils";
+
+const appConfig = RNConfig.getInstance();
+const logger = Logger.getInstance();
+let environmentStateSyncIntervalId: number | null = null;
+
+/**
+ * Fetch the environment state from the backend
+ * @param apiHost - The API host
+ * @param environmentId - The environment ID
+ * @param noCache - Whether to skip the cache
+ * @returns The environment state
+ * @throws NetworkError
+ */
+export const fetchEnvironmentState = async (
+  { apiHost, environmentId }: TJsEnvironmentSyncParams,
+  noCache = false
+): Promise<TJsEnvironmentState> => {
+  const fetchOptions: RequestInit = {};
+
+  if (noCache) {
+    fetchOptions.cache = "no-cache";
+    logger.debug("No cache option set for sync");
+  }
+
+  const url = `${apiHost}/api/v1/client/${environmentId}/environment`;
+
+  const response = await fetch(url, fetchOptions);
+
+  if (!response.ok) {
+    const jsonRes = (await response.json()) as { message: string };
+
+    const error = err({
+      code: "network_error",
+      status: response.status,
+      message: "Error syncing with backend",
+      url: new URL(url),
+      responseMessage: jsonRes.message,
+    });
+
+    // eslint-disable-next-line @typescript-eslint/only-throw-error -- error.error is an Error object
+    throw error.error;
```

Comment:
Can we replace this with `throw error`  as we are not really using resonseMessage from error object and error.error feels kinda wrong, or we can rename the error object

Reply:
actually `error.error` is an Error object which is we throw that only